### PR TITLE
Warn if server is not using G1 garbage collector

### PR DESCRIPTION
**Pull request by dain**

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/server/PrestoSystemRequirements.java [hunk 1]
```diff
@@ -100,6 +106,22 @@ private static void verifyJavaVersion()
         }
     }
 
+    private static void verifyUsingG1Gc()
+    {
+        try {
+            ImmutableList<String> garbageCollectors = ManagementFactory.getGarbageCollectorMXBeans().stream()
```
Comment (electrum) [inline presto-main/src/main/java/com/facebook/presto/server/PrestoSystemRequirements.java]: Declare as `List`

File: presto-main/src/main/java/com/facebook/presto/server/PrestoSystemRequirements.java [hunk 2]
```diff
@@ -100,6 +106,22 @@ private static void verifyJavaVersion()
...
+                    .map(GarbageCollectorMXBean::getName)
+                    .collect(toImmutableList());
+
+            if (garbageCollectors.stream().noneMatch(name -> name.toUpperCase(Locale.US).startsWith("G1 "))) {
+                warnRequirement("Current garbage collector are %s. Presto recommends the G1 garbage collector", garbageCollectors);
```
Comment (electrum) [inline presto-main/src/main/java/com/facebook/presto/server/PrestoSystemRequirements.java]: Add period to end of second sentence

File: presto-main/src/main/java/com/facebook/presto/server/PrestoSystemRequirements.java [hunk 3]
```diff
@@ -100,6 +106,22 @@ private static void verifyJavaVersion()
...
+            }
+        } catch (Exception e) {
```
Comment (electrum) [inline presto-main/src/main/java/com/facebook/presto/server/PrestoSystemRequirements.java]: Can anything here throw checked exceptions? If not, this should be `RuntimeException`

File: presto-main/src/main/java/com/facebook/presto/server/PrestoSystemRequirements.java [hunk 4]
```diff
@@ -100,6 +106,22 @@ private static void verifyJavaVersion()
...
+            // This should never happen since we have verified the OS and JVM above
+            failRequirement("Cannot read garbage collector information");
```
Comment (electrum) [inline presto-main/src/main/java/com/facebook/presto/server/PrestoSystemRequirements.java]: Maybe make this
```java
failRequirement("Cannot read garbage collector information: %s", e);

File: presto-main/src/main/java/com/facebook/presto/server/PrestoSystemRequirements.java (hunk 2 above)
Comment (billonahill) [inline presto-main/src/main/java/com/facebook/presto/server/PrestoSystemRequirements.java]: s/are/is/g

**Pull request closed without merging**